src/main/java/com/scottlogic/librarygradproject/BookService.java

```diff
@@ -0,0 +1,78 @@
+package com.scottlogic.librarygradproject;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Optional;
+
+public class BookService {
+
+    private final BookRepository bookRepo;
+
+    @Autowired
+    public BookService(BookRepository bookRepo) {this.bookRepo = bookRepo;}
+
+    private Book validateBook(Book book) {
+        book.setIsbn(Optional.ofNullable(book.getIsbn()).orElse("").trim());
+        book.setAuthor(Optional.ofNullable(book.getAuthor()).orElseThrow(() -> new IncorrectBookFormatException()).trim());
+        book.setTitle(Optional.ofNullable(book.getTitle()).orElseThrow(() -> new IncorrectBookFormatException()).trim());
+        book.setPublishDate(Optional.ofNullable(book.getPublishDate()).orElse("").trim());
+        String year = book.getPublishDate();
+        if (year.length() > 0) {
+            while (year.length() < 4) {
+                year = "0" + year;
+            }
+        }
+        book.setPublishDate(year);
+        if (!book.getIsbn().matches("|[0-9]{10}|[0-9]{13}") ||
+                (book.getAuthor().length() == 0 || book.getAuthor().length() > 200) ||
+                (book.getTitle().length() == 0 || book.getTitle().length() > 200) ||
+                (book.getPublishDate().length() > 0 && (!book.getPublishDate().matches("|[0-9]{4}") ||
+                        Integer.parseInt(book.getPublishDate()) > Calendar.getInstance().get(Calendar.YEAR)))) {
+            throw new IncorrectBookFormatException();
+        }
+        return book;
+    }
+
+    public List<Book> findAll() {return bookRepo.findAll();}
```

Comment:
Should we try and avoid one-line functions?

Reply:
Do you mean break it down to 3 lines instead?

Reply:
Yes, it does add a couple more lines but it can be a bit clearer